src/mesh/interpolation.cxx

```diff
@@ -65,37 +65,93 @@ const Field3D interp_to(const Field3D &var, CELL_LOC loc)
       // Going between centred and shifted
       
       bindex bx;
-      stencil s;
       CELL_LOC dir; 
       
       // Get the non-centre location for interpolation direction
       dir = (loc == CELL_CENTRE) ? var.getLocation() : loc;
 
       switch(dir) {
       case CELL_XLOW: {
-	start_index(&bx, RGN_NOX);
-	do {
-	  var.setXStencil(s, bx, loc);
-	  result(bx.jx,bx.jy,bx.jz) = interp(s);
-	}while(next_index3(&bx));
+        if (loc == CELL_XLOW) {
+          for(const auto &i : result) {
+            // Producing a stencil centred around a lower X value
+            stencil s;
+            s.p = var[i];
+            s.m = var[i.xm()];
+            s.pp = var[i.xp()];
+            s.mm = var[i.offset(-2,0,0)];
+            result[i] = interp(s);
+          }
+        } else {
+          for(const auto &i : result) {
+            // Producing a stencil centred around a cell centre
+            stencil s;
+            s.p = var[i.xp()];
+            s.m = var[i];
+            s.pp = var[i.offset(2,0,0)];
+            s.mm = var[i.xm()];
+            result[i] = interp(s);
+          }
+        }
 	break;
 	// Need to communicate in X
       }
       case CELL_YLOW: {
-	start_index(&bx, RGN_NOY);
-	do {
-	  var.setYStencil(s, bx, loc);
-	  result(bx.jx,bx.jy,bx.jz) = interp(s);
-	}while(next_index3(&bx));
+        if(var.hasYupYdown() &&
+            ( (&var.yup() != &var) || (&var.ydown() != &var) )) {
+          // Field "var" has distinct yup and ydown fields which
+          // will should be used to calculate interpolation along
+          // the magnetic field
+          throw BoutException("interp_to not implemented for fields with yup/ydown");
+        } else {
+          Field3D var_fa = mesh->toFieldAligned(var);
+          if (loc == CELL_YLOW) {
+            for(const auto &i : result) {
```

Comment:
You need handling of the boundaries.
This way it fails if the `DCHECK>2` and otherwise produces wrong results.

I had the same issue with the aiolos mesh ...

Reply:
Thanks David! This is fixed by e801fc3

Reply:
e801fc3 doe only partially fix the issue:
 * I can still call with `RGN_ALL` to get an error.
 * The bondaries are kept unintialised, which may be ok, but I think STORM relies (or did) on the boundary cells being set. At least my version of STORM wouldn't work reliable, as the recycling depends on the flux on a staggered cell, which should be the first outside the domain. The old code copied the field first, so the boundaries where initialised with the unstaggered values, which hides the bug sufficiently well for most cases. The aiolos has forward and backward stencils, to set the boundaries properly, which I think is what the code should be doing ...